### ISSUE TITLE
Update on getting twitter data from CH

### ISSUE THREAD

**Issue by jamiecmontgomery** [OHI-Science/sb_sense_of_place#2]
An update on where I'm at with twitter data (mainly for record keeping):

It seems that Crimson Hexagon has either updated their querying or I missed this drop down menu previously (the red FILTERS button) but I was able to query tweets that are from Santa Barbara **and** Original Tweets (no retweets). This gives us ~ 5 million. 

![image](https://user-images.githubusercontent.com/6842510/64296125-0f122680-cf28-11e9-9f7e-bba57628f375.png)

I also discovered that I can download more than 10,000 a day, but it requires manual downloads. Originally, I wanted to just download a random 10,000 sample of the entire 5million dataset every day for however many days. But the sample I get today is exactly the same as yesterday. There isn't a clear way to regenerate that random sample. 

The way around this is to get a random sample from a smaller time frame. On average, there are just over 10k tweets in 2 days. So hypothetically I can get almost all of the 5 mill tweets if I manually query every two days (e.g. select Jan 1-2, 2015 then Bulk Export as an excel file). This is really time-consuming but gives us data at a daily scale which I think could actually be quite interesting. 

![image](https://user-images.githubusercontent.com/6842510/64296155-25b87d80-cf28-11e9-98cf-55e80b336532.png)

This downloads a .xls file which, lucky me, is not readable by R even with the `readxl` package! So I then go and change the extension to .xslx and it's readable.

Using the CH data on it's own, I can't get geotag information. So I've used the python library, [`twarc`](https://github.com/DocNow/twarc) to "hydrate" the tweets. The Crimson Hexagon data gives us a unique tweet id for every tweet. The `twarc` library takes that unique ID and gets a LOT more data about that tweet including geotagged coordinates, user information, number of retweets/favorites, and whether or not there is an image attached (which we might want to use).

I downloaded all tweets between Jan 1 and Feb 27 of 2015, and removed those that did not have geotags - leaving me with 5,511 tweets. Starting to really fill up the map:

![image](https://user-images.githubusercontent.com/6842510/64296367-d58deb00-cf28-11e9-925d-20b1dce5d88e.png)

In conclusion - I am getting much more hopeful about using the Crimson Hexagon twitter data now that I've figured out these queries and the python package. I'm going to play around a bit more with the data by taking random 10K samples by month for all months in our dataset (48). 
I think with that dataset we can start actually asking questions of the data!

**Comment by Melsteroni** [comment]
Very cool!!!
